From b4cf7ec68d69761f57ce270b6d6658df8cbf41c3 Mon Sep 17 00:00:00 2001
From: Shilin Victor <chrono.monochrome@gmail.com>
Date: Mon, 20 Aug 2018 22:57:43 +0300
Subject: [PATCH 277/296] Avoid system crash when doing "dumpheap" during disk
 full

This patch add RuntimeException catch at
ActivityThread::handleDumpHeap(), because system server crash happens
when doing "dumpheap" during disk full.
---
 core/java/android/app/ActivityThread.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/java/android/app/ActivityThread.java b/core/java/android/app/ActivityThread.java
index fa389990225..57c9386dd2d 100644
--- a/core/java/android/app/ActivityThread.java
+++ b/core/java/android/app/ActivityThread.java
@@ -5246,7 +5246,7 @@ public final class ActivityThread {
         if (dhd.managed) {
             try {
                 Debug.dumpHprofData(dhd.path, dhd.fd.getFileDescriptor());
-            } catch (IOException e) {
+            } catch (IOException | RuntimeException e) {
                 Slog.w(TAG, "Managed heap dump failed on path " + dhd.path
                         + " -- can the process access this path?");
             } finally {
-- 
2.11.0

